content/blog/full-coverage-of-azure-resources-with-azure-native/index.md

```diff
@@ -0,0 +1,281 @@
+   "github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+   pulumi.Run(func(ctx *pulumi.Context) error {
+       resourceGroup, err := resources.NewResourceGroup(ctx, "resourceGroup", nil)
+       if err != nil {
+           return err
+       }
+
+       account, err := storage.NewStorageAccount(ctx, "sa", &storage.StorageAccountArgs{
+           ResourceGroupName: resourceGroup.Name,
+           Sku: &storage.SkuArgs{
+               Name: storage.SkuName_Standard_LRS,
+           },
+           Kind: storage.KindStorageV2,
+       })
+
+       return err
+   })
+}
+```
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+The top-level resources will use stable API versions and take no arbitrary breaking changes within a given major version of the package once we reach general availability; during the preview, breaking changes may appear in `0.x` minor versions.
+
+We still provide access to every version of every ARM API if you want to use a brand-new version or a specific version from the past. Each API version is mapped to its own module or namespace within the resource provider.
+
+### Auto-Naming
+
+The original preview of the native Azure provider (then called “Azure NextGen”) didn’t yet support auto-naming, a common feature of other Pulumi providers. Now, it provides full support for auto-naming. Azure resources will get a random suffix in the name by default. In the following example, Pulumi will create a resource group called, for instance, `rg53acf911`.
+
+{{< chooser language "typescript,python,csharp,go" >}}
+
+{{< choosable language typescript >}}
+
+```typescript
+import * as azure from "@pulumi/azure-native";
+const resourceGroup = new azure.resources.ResourceGroup("rg");
+```
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+```python
+import pulumi_azure_native as azure
+resource_group = azure.resources.ResourceGroup('rg')
+```
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+```csharp
+using Pulumi;
+using Pulumi.AzureNative.Resources;
+class MyStack : Stack
+{
+   public MyStack()
+   {
+       var resourceGroup = new ResourceGroup("rg");
+   }
+}
+```
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+```go
+package main
+
+import (
+   resources "github.com/pulumi/pulumi-azure-native/sdk/go/azure/resources"
+   "github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+   pulumi.Run(func(ctx *pulumi.Context) error {
+       _, err := resources.NewResourceGroup(ctx, "resourceGroup", nil)
+       return err
+   })
+}
+```
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+Pass an explicit value to the `resourceGroupName` property if you want to control the resource's exact name.
+
+However, suppose you create a sub-resource, e.g., a container under a storage account, a database under a database account, or a slot under a web app. In that case, Pulumi will not append any random suffix because the names of those resources don’t have to be globally unique.
```

Comment:
```suggestion
However, suppose you create a sub-resource, e.g., a container under a storage account, a database under a database account, or a slot under a web app. In that case, Pulumi will not append a random suffix because those resources names don’t have to be globally unique.
```